### ISSUE TITLE
Progress reporting

### ISSUE THREAD

**Issue by hayellis** [ddeboer/DdeboerDataImportBundle#9]
Hi,

I'we looked at DdeboerDataImportBundle and find it to be very well thought out. I'we been working on a similar component for one of my project (with rather bad results). Your implementation fills all my requirements elegantly, but with one exception.

I need to be able to get continuous progress reports. As an import could take forever, it's critical to always give feedback to the user on what is being done, and eventual errors that might pop up. I also want to use the same code but with a different "progress report output adapter", one for the console, and one for the web.

The console is self explained, but for the web, i would like to make use of symfonys StreamedResponse, and output a stream of JSON, in a way i can make use of http://code.google.com/p/jquery-stream/ and update the user interface as the import runs.

Using the symfony console OutputInterface is way to centered around human readable line-by-line output, and do not allow me to pass information useful to my use-case.

The general idea with my web output is to:
1. Setup readers, writers, and workflow
2. Pass the workflow to a StreamedRespons callback (that is executed right before symfony exit stage)
3. Run the workflow with my json stream adapter from the callback, and have it output the progress as it happens.

Sorry, if I'm a bit unclear.

**Comment by ddeboer** [comment]
Sounds like a great idea. But is what you want not already possible?

Currently, the `Workflow` calls the `writeItem` method on each `Writer` for each data item. So you could create, for instance, a `StreamedResponseWriter`. You would then pass the `StreamedResponse` to that class and have the writer output any progress to the streamed response. Does that make sense?

**Comment by hayellis** [comment]
I was looking for some kind of dedicated output, so that i totally overlooked the generic nature of the Writer-concept. Really good idea. :)

I'll be experimenting with that approach. Thanks.

**Comment by ddeboer** [comment]
Cool. Let me know if using the writers doesn't work out for you after all. And if you want to contribute your `StreamedResponseWriter` to the bundle, you're welcome to do so. :)

**Comment by hayellis** [comment]
I did some playing around and came up with this gist: https://gist.github.com/3164137

I'we used your example from the README.md. 

I'we used the standardized EventSource protocol.

Add `"igorw/event-source": "*"` to composer.json for the EventSource class.

I thought of creating a special `StreamedResponseWorkflow` class but figured it would add nothing. But it would be nice to hide away the `$eventSourceWriter->getStream()->getHeaders()` part from the response construct.

As part of a proper progress report mechanism, one would have to add stuff like initial counting of items and pass that to the workflow and writer. But i can see how that would be outside the scope of the Writer.
One could inject the workflow into the writers and let the developer decide if it's interesting or not. Like you see, i start to output stuff in the `prepare` function to demonstrate.
